src/library_pthread.js

```diff
@@ -278,8 +278,6 @@ var LibraryPThread = {
         } else if (cmd === 'detachedExit') {
 #if ASSERTIONS
           assert(worker.pthread);
-          var detach_state = Atomics.load(HEAPU32, (worker.pthread.threadInfoStruct + {{{ C_STRUCTS.pthread.detach_state }}}) >> 2);
-          assert(detach_state == {{{ cDefine('DT_EXITED') }}});
 #endif
```

Comment:
Why not leave this assertion?

Reply:
Removing this means the JS no no more need to know about detach_state (keeping the logic all in native code simplifies the model IMHO).  Also, detachedExit has exactly one callsite which is in emscripten_thread_exit so it seems little unnecessary.   I think it will be good if in the long run we can move all the threadInfoStruct handling to native code and reduce the JS code to just worker handling.